suites/apis/googleDataAccessTest.js

```diff
@@ -0,0 +1,309 @@
+  console.log(`Running: ${fenceCmd}`)
+  response = bash.runCommand(fenceCmd, 'fence');
+
+  console.log('Clean up Google Bucket Access Groups from previous runs...\n');
+  bash.runJob('google-verify-bucket-access-group');
+});
+
+After(async (fence, users) => {
+  // Cleanup after each scenario
+  const unlinkResults = Object.values(users).map(user => fence.do.unlinkGoogleAcct(user));
+  await Promise.all(unlinkResults);
+});
+
+AfterSuite(async (fence, users) => {
+  console.log(`Running usersync job`);
+  bash.runJob('usersync');
+});
+
+Scenario('test google data access via usersync: usersync, Google link, generate temp creds, bucket access, usersync access file 2, bucket access, delete temp creds @reqGoogle @googleDataAccess',
+  async (fence, users, google) => {
+  console.log('make sure users google accounts are unlinked');
+  await fence.complete.forceUnlinkGoogleAcct(users.user0);
+  await fence.complete.forceUnlinkGoogleAcct(users.user1);
+  await fence.complete.forceUnlinkGoogleAcct(users.user2);
+
+  console.log('linking users google accounts');
+  await fence.complete.linkGoogleAcctMocked(users.user0);
+  await fence.complete.linkGoogleAcctMocked(users.user1);
+  await fence.complete.linkGoogleAcctMocked(users.user2);
+
+  console.log(`creating temporary google creds for users with usernames:  ${users.user0.username}, ${users.user1.username}, ${users.user2.username}`);
+  // call our endpoint to get temporary creds
+  // NOTE: If this fails and you don't know why, it *might* be that we've hit our limit
+  //       for SA keys on this user's Google Service Account. They *should* be cleaned
+  //       up after every test but if they aren't, you need to delete the keys from the
+  //       fence database (in table google_service_account_keys) AND delete them from
+  //       the Google Cloud Platform. Check usersUtil.js for information about these users
+  //       (specifically their username is their Google Account email, you can use that
+  //        to find their service account in the GCP)
+  const tempCreds0Res = await fence.complete.createTempGoogleCreds(
+    users.user0.accessTokenHeader);
+  const tempCreds1Res = await fence.complete.createTempGoogleCreds(
+    users.user1.accessTokenHeader);
+  const tempCreds2Res = await fence.complete.createTempGoogleCreds(
+    users.user2.accessTokenHeader);
+
+  console.log('saving temporary google creds to file');
+  const creds0Key = tempCreds0Res.body.private_key_id;
+  const creds1Key = tempCreds1Res.body.private_key_id;
+  const creds2Key = tempCreds2Res.body.private_key_id;
+  const pathToCreds0KeyFile = creds0Key + '.json';
+  const pathToCreds1KeyFile = creds1Key + '.json';
+  const pathToCreds2KeyFile = creds2Key + '.json';
+
+  fs.writeFile(pathToCreds0KeyFile, JSON.stringify(tempCreds0Res.body), function(err) {
+    if (err) {
+      return console.log(err);
+    }
+    console.log(`Google creds file ${pathToCreds0KeyFile} saved!`);
+  });
```

Comment:
there is a util function that makes sure to wait for the file to be written before returning:
`await files.createTmpFile(pathToCreds0KeyFile, JSON.stringify(tempCreds0Res.body));`

Reply:
oh, nice. thanks!